polaris-react/src/components/Choice/Choice.scss

```diff
@@ -9,6 +9,10 @@
   @media #{$p-breakpoints-sm-up} {
     display: inline-flex;
   }
+
+  #{$se23} & {
+    align-items: center;
+  }
```

Comment:
Align checkbox to center of label

|Before|After|
|-|-|
|<img width="142" alt="Screenshot 2023-06-09 at 11 15 44 AM" src="https://github.com/Shopify/polaris/assets/20652326/680d6560-6b2b-43a3-814b-5ab8864c2ec8">|<img width="141" alt="Screenshot 2023-06-09 at 11 15 12 AM" src="https://github.com/Shopify/polaris/assets/20652326/fd224128-003e-4019-b166-0d425720b8b0">|